### DISCUSSION TITLE
Downloads

### DISCUSSION THREAD

**Discussion by rmccar** [ONSdigital/design-system#716] [Feedback]
Use this place to discuss [Downloads](https://ons-design-system.netlify.com/components/downloads/) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by trevorsaint** [comment]
**An improvement for this component would be to:**

Add `tabindex="-1"` for the image thumbnail link.

We have an `aria-hidden="true"` attribute, which is great because we don’t want screen readers reading out the link twice.

However a keyboard user would still have to tab twice for the thumbnail link and the heading link.

They do the same thing. Adding `tabindex="-1"` will resolve this.

**Comment by rmccar** [comment]
> **An improvement for this component would be to:**
> 
> Add `tabindex="-1"` for the image thumbnail link.
> 
> We have an `aria-hidden="true"` attribute, which is great because we don’t want screen readers reading out the link twice.
> 
> However a keyboard user would still have to tab twice for the thumbnail link and the heading link.
> 
> They do the same thing. Adding `tabindex="-1"` will resolve this.

Sounds good Ive raised a PR to fix this

**Comment by trevorsaint** [comment]
We need to update the thumbnail path to allow for a smallPath and a largePath.

Currently the macro expects a small and large folder (hardcoded into macro). However, with the way Craft does image modifications it names the folders with an underscore.

Allowing for a smallPath and a largePath would sort this issue out and also make the component more flexible.

**Comment by trevorsaint** [comment]
We need the ability to have a fallback image if no image specified as part of the component.

**Comment by trevorsaint** [comment]
I have spotted a bug with this component when used inside `page__body`. The margin is inherited causing incorrect margins.

**See example:**

<img width="658" alt="Screenshot 2021-01-12 at 17 32 06" src="https://user-images.githubusercontent.com/4989027/104350716-618c2b80-54fc-11eb-8dea-5ac83ac50e12.png">

The top image is how it should look. The bottom image is how it currently looks.

It might be worth adding a title class for the heading to control how it should be rendered.

**Comment by jrbarnes9** [comment]
Download thumbnail images shouldn't require `alt` text, as they are decorative
